exporters/common/src/test/java/io/opentelemetry/exporter/internal/TlsUtilTest.java

```diff
@@ -60,4 +78,39 @@ void generatePrivateKey_Invalid() {
         .isInstanceOf(SSLException.class)
         .hasMessage("Unable to generate key from supported algorithms: [EC]");
   }
+
+  /**
+   * Append <a href="https://datatracker.ietf.org/doc/html/rfc7468#section-5.2">explanatory text</a>
+   * prefix and verify {@link TlsUtil#keyManager(byte[], byte[])} succeeds.
+   */
+  @ParameterizedTest
+  @MethodSource("keyManagerArgs")
+  void keyManager_CertWithExplanatoryText(SelfSignedCertificate selfSignedCertificate)
+      throws IOException {
+    Path certificate = tempDir.resolve("certificate");
+    Files.write(certificate, EXPLANATORY_TEXT.getBytes(StandardCharsets.UTF_8));
+    Files.write(
+        certificate,
+        com.google.common.io.Files.toByteArray(selfSignedCertificate.certificate()),
+        StandardOpenOption.APPEND);
+    Files.write(
+        certificate,
+        "\n".getBytes(StandardCharsets.UTF_8),
+        StandardOpenOption.APPEND);
```

Comment:
If I understand correct, the approach in this PR of using `factory.generateCertificates(is)` is able to handle a trailing line break where as the approach on `main` can not?

Reply:
yes, that is correct.